controllers/namespace_reconciler.go

```diff
@@ -0,0 +1,90 @@
+package controllers
+
+import (
+	v1 "alexellis/registry-creds/api/v1"
+	"context"
+	"fmt"
+	"strings"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type NamespaceReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// Reconcile applies a PullSecret to a Namespace
+func (r *NamespaceReconciler) Reconcile(pullSecret v1.ClusterPullSecret, namespaceName string) error {
```

Comment:
Why do we need this? We already have a NamespaceWatcher that applies secrets by calling `r.SecretReconciler.Reconcile`

The problem I am seeing with the changes in this PR is that they are not defensible - i.e. you haven't proven and existing issue with reproduction instructions, if you could do that, then these changes would be easer to justify.

Reply:
Try to come up with some scenarios that prove why the existing code doesn't work.